scripts/utils/migration_scripts/collectionsMigration/main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type config struct {
+	teamsDir,
+	collectionDir,
+	collectionCopyDir string
+	migration bool
+}
+
+type teamCollection struct {
+	id   string
+	name string
+}
+
+type collection struct {
+	InProgressUris        []interface{} `json:"inProgressUris"`
+	CompleteUris          []interface{} `json:"completeUris"`
+	ReviewedUris          []interface{} `json:"reviewedUris"`
+	PendingDeletes        []interface{} `json:"pendingDeletes"`
+	PublishResults        []interface{} `json:"publishResults"`
+	TimeseriesImportFiles []interface{} `json:"timeseriesImportFiles"`
+	PublishTransactionIds struct {
+	} `json:"publishTransactionIds"`
+	EventsByUri struct {
+	} `json:"eventsByUri"`
+	Datasets        []interface{} `json:"datasets"`
+	Interactives    []interface{} `json:"interactives"`
+	DatasetVersions []interface{} `json:"datasetVersions"`
+	ApprovalStatus  string        `json:"approvalStatus"`
+	PublishComplete bool          `json:"publishComplete"`
+	IsEncrypted     bool          `json:"isEncrypted"`
+	Events          []struct {
+		Date  time.Time `json:"date"`
+		Type  string    `json:"type"`
+		Email string    `json:"email"`
+	} `json:"events"`
+	Id    string   `json:"id"`
+	Name  string   `json:"name"`
+	Type  string   `json:"type"`
+	Teams []string `json:"teams"`
+}
+
+func readConfig() *config {
+	conf := &config{}
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		switch pair[0] {
+		case "teamsDir":
+			missingVariables("teamsDir", pair[1])
+			conf.teamsDir = pair[1]
+		case "collectionDir":
+			missingVariables("collectionDir", pair[1])
+			conf.collectionDir = pair[1]
+		case "collectionCopyDir":
+			missingVariables("collectionCopyDir", pair[1])
+			conf.collectionCopyDir = pair[1]
+		case "migration":
+			missingVariables("migration", pair[1])
+			boolValue, err := strconv.ParseBool(pair[1])
+			if err != nil {
+				log.Fatal(err)
+			}
+			conf.migration = boolValue
+		}
+	}
+	return conf
+}
+
+func missingVariables(envValue string, value string) bool {
+	if len(value) == 0 {
+		log.Fatalf("Please set Environment Variable: %s", envValue)
+	}
+	return true
+}
+
+func getDirList(dir string) ([]fs.FileInfo, string, error) {
+	path, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, "", err
+	}
+
+	tmpdirFilesList, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, path, err
+	}
+	var dirFilesList []fs.FileInfo
+
+	for _, f := range tmpdirFilesList {
+		if strings.Contains(f.Name(), ".json") {
+			dirFilesList = append(dirFilesList, f)
+		}
+	}
+
+	return dirFilesList, path, nil
+}
+
+func getTeamsList(dir string) []teamCollection {
```

Comment:
Instead of returning a slice of teams, you could instead return two maps (teamids,teammaps`map[string]string`). 
This would make the the lookups much simpler in the amend/revert functions. (see comments below)